pkg/ccl/backupccl/manifest_handling.go

```diff
@@ -1259,38 +1274,60 @@ func readLatestCheckpointFile(
 	// First try reading from the progress directory. If the backup is from
 	// an older version, it may not exist there yet so try reading
 	// in the base directory if the first attempt fails.
-	r, err := exportStore.ReadFile(ctx, backupProgressDirectory+"/"+filename)
-	if err != nil {
-		r, err = exportStore.ReadFile(ctx, filename)
-		if err != nil {
-			return nil, errors.Wrapf(err, "%s could not be read in the base or progress directory", filename)
-		}
+	var checkpoint string
+	var checkpointFound bool
+	var r ioctx.ReadCloserCtx
+	var err error
+
+	// We name files such that the most recent checkpoint will always
+	// be at the top, so just grab the first filename.
+	err = exportStore.List(ctx, backupProgressDirectory, "", func(p string) error {
+		p = strings.TrimPrefix(p, "/")
+		checkpoint = strings.TrimSuffix(p, backupManifestChecksumSuffix)
```

Comment:
nit: comment here that the first file returned by List could be the checkpoint or its checksum but we're only concerned with its timestamped prefix. We resolve the checksum and checkpoint separately below.